server/server.js

```diff
@@ -370,7 +371,107 @@ function startServer() {
   );
 }
 
+async function fetchCitybikeSeasons() {
+  const client = new CosmosClient(process.env.CITYBIKE_DB_CONN_STRING);
+  const database = client.database(process.env.CITYBIKE_DATABASE);
+  const container = database.container('schedules');
+  const query = {
+    query: 'SELECT * FROM c',
+  };
+
+  const { resources } = await container.items.query(query).fetchAll();
+  return resources;
+}
+
+const parseDate = (year, month, day) =>
+  // eslint-disable-next-line radix
+  new Date(parseInt(year), parseInt(month) - 1, parseInt(day));
+
+function buildCitybikeConfig(cb, networkName, configName) {
+  const inSeason = cb.inSeason.split('-');
+  const [startDay, startMonth, startYear] = inSeason[0].split('.');
+  const [endDay, endMonth, endYear] = inSeason[1].split('.');
+  const [preDay, preMonth, preYear] = cb.preSeason.split('.');
+  return {
+    config: configName,
+    networkName,
+    region: cb.region,
+    enabled: cb.enabled,
+    season: {
+      preSeasonStart: parseDate(preYear, preMonth, preDay),
+      start: parseDate(startYear, startMonth, startDay),
+      end: parseDate(endYear, endMonth, endDay),
+    },
+  };
+}
+
+function handleCitybikes(schedules, configName) {
+  const bikes = schedules.filter(
+    cb => cb.region.toLowerCase() === configName || cb.config === configName,
+  );
+  const cbConfigurations = [];
+  bikes.forEach(bike => {
+    // This check is due to naming scheme in HSL citybike networks (Smoove and Vantaa)
+    if (bike.config && bike.config === 'hsl') {
+      if (bike.region === 'Vantaa') {
+        cbConfigurations.push(
```

Comment:
Is Vantaa the only use case for region? Do we need region at all? The hard coded config names hsl and vantaa are really bad, so could this be like:

const networkName = seasonDef.networkName || operator.concat('_', configName);

This oneliner seems to replace 10 + lines of if else blocks.

Reply:
Region is defined in every other citybike cities name: For example there is multiple cities that uses same operator (For example donkey). And in waltti configurations they are named operator_city For example donkey_kotka. Only vantaa configuration is called vantaa, and hsl is called smoove, wihtout any regions. Hence the exceptions.

But You are right about the naming. Will fix them.